tensordict/tensorclass.py

```diff
@@ -721,20 +755,29 @@ def wrapper(cls, tensordict, non_tensordict=None):  # noqa: D417
                     raise KeyError(
                         f"{key} is present in both tensor and non-tensor dicts."
                     )
-        # bypass initialisation. this means we don't incur any overhead creating an
-        # empty tensordict and writing values to it. we can skip this because we already
-        # have a tensordict to use as the underlying tensordict
-        tc = cls.__new__(cls)
-        tc.__dict__["_tensordict"] = tensordict
-
-        tc.__dict__["_non_tensordict"] = (
-            non_tensordict if non_tensordict is not None else {}
-        )
-        # since we aren't calling the dataclass init method, we need to manually check
-        # whether a __post_init__ method has been defined and invoke it if so
-        if hasattr(tc, "__post_init__"):
-            tc.__post_init__()
-        return tc
+        if not torch.compiler.is_dynamo_compiling():
+            # bypass initialisation. this means we don't incur any overhead creating an
+            # empty tensordict and writing values to it. we can skip this because we already
+            # have a tensordict to use as the underlying tensordict
+            tc = cls.__new__(cls)
+            tc.__dict__["_tensordict"] = tensordict
+            tc.__dict__["_non_tensordict"] = (
+                non_tensordict if non_tensordict is not None else {}
+            )
+            # since we aren't calling the dataclass init method, we need to manually check
+            # whether a __post_init__ method has been defined and invoke it if so
+            if hasattr(tc, "__post_init__"):
+                tc.__post_init__()
+            return tc
+        else:
+            # TODO: things that did NOT work: **tensordict, dict(tensordict)
```

Comment:
@lezcano @anijain2305 

All of these (`dict(tensordict)` or `**tensordict`) are valid syntaxes but only `dict(tensordict.items())` worked